frontend/ui/renderimage.lua

```diff
@@ -66,6 +67,13 @@ end
 -- @treturn BlitBuffer
 function RenderImage:renderImageDataWithMupdf(data, size, width, height)
     if not Mupdf then Mupdf = require("ffi/mupdf") end
+    -- NOTE: Kobo's fb is BGR, not RGB. Handle the conversion in MuPDF if needed.
+    if Mupdf.bgr == nil then
+        Mupdf.bgr = false
+        if Device:isKobo() then
+            Mupdf.bgr = true
```

Comment:
May be we shouldn't harcode `:isKobo()` here and elsewhere, and use another function `Device:hasBGRScreen()`, or `Screen:isBGR()`. Would be easier to adapt or disable.

Reply:
That sounds like a decent design decision, yeah, will look into it tomorrow ;).